cvmfs/mountpoint.cc

```diff
@@ -1455,6 +1455,53 @@ bool MountPoint::CreateDownloadManagers() {
     download_mgr_->ShardProxies();
   }
 
+  // configure http tracing header
+  if (options_mgr_->GetValue("CVMFS_HTTP_TRACING", &optarg) &&
+      options_mgr_->IsOn(optarg)) {
+    download_mgr_->EnableHTTPTracing();
+    if (options_mgr_->GetValue("CVMFS_HTTP_TRACING_HEADERS", &optarg)) {
+      std::vector<std::string> tokens = SplitString(optarg, '|');
+      sanitizer::AlphaNumSanitizer sanitizer;
+
+      for (unsigned int i = 0; i < tokens.size(); i++) {
+        std::string token = Trim(tokens[i]);
+
+        std::vector<std::string> key_val = SplitString(token, ':');
+
+        if (key_val.size() != 2) {
+          LogCvmfs(kLogCvmfs, kLogSyslog | kLogDebug,
+             "Http tracing header: Skipping current token part of "
+             "CVMFS_HTTP_TRACING_HEADERS! Invalid "
+             "<key:value> pair. Token: %s", token.c_str());
+          continue;
+        }
+
+        std::string prefix = "X-CVMFS-";
+        std::string key = Trim(key_val[0]);
+
+        if (!sanitizer.IsValid(key)) {
```

Comment:
Let's also set a maximal length (there may be something on length in the HTTP standard, otherwise: 100B?)

Reply:
https://stackoverflow.com/questions/686217/maximum-on-http-header-values old article but smallest size is 4k for *sum of the request line and ALL header fields*. so i would say max 200 chars?